crates/stc_ts_file_analyzer/src/analyzer/expr/call_new.rs

```diff
@@ -2932,6 +2932,11 @@ impl Analyzer<'_, '_> {
                     }
 
                     if arg.spread.is_some() {
+                        if let Err(err) = self.get_iterator_element_type(arg.span(), Cow::Borrowed(&arg.ty), false, Default::default()) {
+                            report_err!(Error::SpreadMustBeTupleOrPassedToRest { span: arg.span() });
+                            continue;
+                        }
+
                         match arg.ty.normalize() {
```

Comment:
What I mentioned in the original issue was this part.

Reply:
The result of `get_iterator_element_type` should be used to return `Ok(())` in the case where element assignment success.

Refer to https://github.com/dudykr/stc/blob/941a9a4981f2d07d84b8e24cc4256576b0e1ac23/crates/stc_ts_file_analyzer/src/analyzer/expr/call_new.rs#L2943

Reply:
Made the changes. Is that what you meant?

Reply:
Yes, thank you!